parallel-consumer-core/src/test-integration/java/io/confluent/parallelconsumer/integrationTests/state/PartitionStateCommittedOffsetIT.java

```diff
@@ -0,0 +1,524 @@
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.AlterConfigOp;
+import org.apache.kafka.clients.admin.ConfigEntry;
+import org.apache.kafka.clients.consumer.*;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.KafkaFuture;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.config.ConfigResource;
+import org.apache.kafka.common.config.TopicConfig;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+import org.testcontainers.containers.KafkaContainer;
+import org.testcontainers.shaded.org.awaitility.Awaitility;
+import pl.tlinkowski.unij.api.UniLists;
+import pl.tlinkowski.unij.api.UniMaps;
+import pl.tlinkowski.unij.api.UniSets;
+
+import java.time.Duration;
+import java.util.*;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Collectors;
+
+import static com.google.common.truth.Truth.assertThat;
+import static com.google.common.truth.Truth.assertWithMessage;
+import static io.confluent.csid.utils.JavaUtils.getLast;
+import static io.confluent.parallelconsumer.ParallelConsumerOptions.ProcessingOrder.PARTITION;
+import static io.confluent.parallelconsumer.ParallelConsumerOptions.ProcessingOrder.UNORDERED;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static pl.tlinkowski.unij.api.UniLists.of;
+
+/**
+ * Integration test versions of {@link io.confluent.parallelconsumer.state.PartitionStateCommittedOffsetTest}, where
+ * committed offset gets moved around or deleted, or random offsets are removed.
+ *
+ * @author Antony Stubbs
+ * @see io.confluent.parallelconsumer.state.PartitionStateCommittedOffsetTest
+ */
+@Slf4j
+class PartitionStateCommittedOffsetIT extends BrokerIntegrationTest<String, String> {
+
+    public static final OffsetResetStrategy DEFAULT_OFFSET_RESET_POLICY = OffsetResetStrategy.EARLIEST;
+    AdminClient ac;
+
+    String groupId;
+
+    ParallelEoSStreamProcessor<String, String> pc;
+
+    TopicPartition tp;
+
+    int TO_PRODUCE = 200;
+
+    private OffsetResetStrategy offsetResetStrategy = DEFAULT_OFFSET_RESET_POLICY;
+
+    @BeforeEach
+    void setup() {
+        setupTopic();
+        tp = new TopicPartition(getTopic(), 0);
+        groupId = getKcu().getConsumer().groupMetadata().groupId();
+        this.ac = getKcu().getAdmin();
+    }
+
+    /**
+     * Test for offset gaps in partition data (i.e. compacted topics)
+     */
+    @Test
+    void compactedTopic() {
+        KafkaContainer compactingBroker = setupCompactingKafkaBroker();
```

Comment:
compacting broker seems to have introduced a lot of timeout errors:

11:26:10  26:10.195  WARN  [kafka-producer-network-thread | producer-164]  (NetworkClient.java:775)#processDisconnection [Producer clientId=producer-164] Connection to node 1 (/172.31.15.245:34928) could not be established. Broker may not be available.

Might be connected to having multiple brokers running and KCU not closing cleanly to a broker that's gone?

Reply:
fixed with tighter resource management and locking down kcu access